### DISCUSSION TITLE
Supabase integration ( Vercel )

### DISCUSSION THREAD

**Discussion by ousszizou** [supabase/supabase#3918] [Questions] [answered]
Hey folks,
I don't know why supabase can't fetch my vercel's projects !! I can't figure where is the problem exactly.
Even I tried this template here: https://github.com/vercel/nextjs-subscription-payments & didn't works.

Error message: Retrieve vercel projects failed. An error has occured: Failed to fetch.

**Comment by natedunn** [comment]
Also getting this error.
![Screen Shot 2021-11-16 at 8 09 48 PM](https://user-images.githubusercontent.com/5555138/142096369-20572eb6-e07f-451a-ad15-7a05bcffce24.png)

**Comment by ousszizou** [comment]
I tried once again & the error message changed: Retrieve vercel token failed.

**Comment by MildTomato** [comment]
@ousszizou @natedunn 
Thanks for posting this, we'll need to investigate.

**Reply by natedunn** [reply]
thanks!

**Reply by ousszizou** [reply]
thanks!
I will wait ^__^

**Reply by MildTomato** [reply]
sorry for delay on this - we've identified the problem. working on a fix right now

**Comment by natedunn** [comment]
@MildTomato Here is an updated error that @ousszizou was mentioning
![Screen Shot 2021-11-18 at 7 58 22 PM](https://user-images.githubusercontent.com/5555138/142551838-fce3e9a9-0097-4cd2-9633-56db85f314ea.png)

**Reply by ousszizou** [reply]
thanks for adding these images <3 it describes the error better.

**Comment by natedunn** [comment]
@MildTomato This might not be helpful as you might already know what is happening behind the scenes but in any case here is an update on my end. I am now getting a 404 error instead of the screen linked above.

![Screen Shot 2021-11-23 at 3 10 02 PM](https://user-images.githubusercontent.com/5555138/143129507-22d2497b-ea3a-404c-8182-e73df35b1fac.png)

**Reply by ousszizou** [reply]
the same here, I think they are working on it

**Reply by natedunn** [reply]
yeah very likely

**Reply by MildTomato** [reply]
sorry for delay on this - we've identified the problem. working on a fix right now

**Reply by natedunn** [reply]
@MildTomato Thanks for the update!

**Reply by benthillerkus** [reply]
> sorry for delay on this - we've identified the problem. working on a fix right now

Can I create a new Supabase project now and connect with Vercel later?

**Comment by carlosstenzel** [comment]
Associated https://github.com/supabase/supabase/issues/3982

the problem is related to url, you need to manually configure the .env file, and commit to github

**Reply by ousszizou** [reply]
Could you please show how we should configure it in .env file ?

**Reply by carlosstenzel** [reply]
When starting the project in vercel, he already cloned the repository, and configured the automatic deploy (probably) in your github.

Just create the file at the root (.env) , according to your project.

**Reply by carlosstenzel** [reply]
For exemple 
https://github.com/vercel/nextjs-subscription-payments

create file .env in root, and change the information 


```
# Update these with your Supabase details from your project settings > API
NEXT_PUBLIC_SUPABASE_URL=https://your-project.supabase.co
NEXT_PUBLIC_SUPABASE_ANON_KEY=
SUPABASE_SERVICE_ROLE_KEY=

# Update these with your Stripe credentials from https://dashboard.stripe.com/apikeys
NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY=pk_test_1234
STRIPE_SECRET_KEY=sk_test_1234
STRIPE_WEBHOOK_SECRET=whsec_1234
```

**Reply by natedunn** [reply]
to be clear: do **_not_** commit .env to your repo.

**Reply by carlosstenzel** [reply]
can be set in vercel the variables too

**Reply by carlosstenzel** [reply]
in a private repo i don't see problems. now if it's public, then I think it's better not.

**Reply by carlosstenzel** [reply]
if you change the variavies in vercel. it is necessary to do the rebuild by the platform

**Reply by natedunn** [reply]
> in a private repo i don't see problems. now if it's public, then I think it's better not.

you do you. but if there is _ever_ a chance that the repo will be public or accessible broadly, then do not. 

best practice is to _never_ do it, though.

**Reply by natedunn** [reply]
but as you mentioned, setting variables in Vercel is there for a reason. to prevent this issue.

**Comment by hugo-dystech** [comment]
Hey guys, just to add on top of this issue same problem here
<img width="794" alt="Screen Shot 2021-11-25 at 8 18 47 am" src="https://user-images.githubusercontent.com/58765031/143314127-cb83c984-411b-4c2c-a451-0b871cfbc514.png">

**Reply by carlosstenzel** [reply]
you need close a window (supabase). and insert values ​​from the .env file into the variables ​​manually in settings and do the rebuild (in vercel)

**Reply by hugo-dystech** [reply]
How do you mean "do the rebuild"?

**Reply by carlosstenzel** [reply]
sorry it's redeploy 

Deployments -> select commit -> 
![Screen Shot 2021-11-24 at 19 23 09](https://user-images.githubusercontent.com/3890516/143320320-14d6d595-98a3-4e9d-80c4-c1fd523bb517.png)

**Comment by carlosstenzel** [comment]
![Screen Shot 2021-11-24 at 18 37 47](https://user-images.githubusercontent.com/3890516/143316071-a86d5f4f-cff6-4218-8a05-8ee3a5ecf955.png)
## check integration permissions
![Screen Shot 2021-11-24 at 18 38 40](https://user-images.githubusercontent.com/3890516/143316077-732d5e34-2d53-4e31-bcdd-a5dd5619b0b2.png)

**Comment by carlosstenzel** [comment]
![Screen Shot 2021-11-24 at 18 41 44](https://user-images.githubusercontent.com/3890516/143316361-092d0fcd-2f7c-40d4-8a7d-6c7be7daf4ad.png)

**Comment by MildTomato** [comment, marked as answer]
@ousszizou @carlosstenzel @natedunn 

we have deployed some changes lately - the issues mentioned in this thread seem to be working again.

**Reply by ousszizou** [reply]
works like a charm <3 big thanks

**Comment by sjoukedv** [comment]
I'm still experiencing issues with `https://api.supabase.io/platform/vercel/projects` resulting in a `401`. I have both my Gitlab and Github credentials added to Vercel, but it still doesn't work

**Reply by joserobleda** [reply]
Same here!

**Reply by sjoukedv** [reply]
You can manually add the details from Supabase 'Settings' > 'API' and add the following to your environment in Vercel:

```
NEXT_PUBLIC_SUPABASE_URL=
NEXT_PUBLIC_SUPABASE_ANON_KEY=
SUPABASE_SERVICE_ROLE_KEY=
```

This at least got me started.

**Reply by joserobleda** [reply]
Thanks, works for me :)
